runtime/src/iree/tooling/context_util.c

```diff
@@ -495,6 +495,42 @@ iree_status_t iree_tooling_resolve_modules(
   return status;
 }
 
+iree_status_t iree_tooling_find_single_exported_function(
+    iree_vm_module_t* module, iree_vm_function_t* out_function) {
+  memset(out_function, 0, sizeof(*out_function));
+  iree_vm_module_signature_t module_signature =
+      iree_vm_module_signature(module);
+  iree_host_size_t exported_functions = 0;
+  for (iree_host_size_t i = 0; i < module_signature.export_function_count;
+       ++i) {
+    iree_vm_function_t function = {0};
+    IREE_RETURN_IF_ERROR(
+        iree_vm_module_lookup_function_by_ordinal(
+            module, IREE_VM_FUNCTION_LINKAGE_EXPORT, i, &function),
+        "looking up function export %zu", i);
+    iree_string_view_t function_name = iree_vm_function_name(&function);
+    if (iree_string_view_starts_with(function_name,
+                                     iree_make_cstring_view("__")) ||
+        iree_string_view_find_char(function_name, '$', 0) !=
+            IREE_STRING_VIEW_NPOS) {
+      // Skip internal or special functions.
```

Comment:
Micro nit/up to you: could this be before line 512? This would make the conditional being checked easier to follow for me.